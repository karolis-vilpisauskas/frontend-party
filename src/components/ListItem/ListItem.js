import React from "react";
import PropTypes from "prop-types";
import "./ListItem.scss";

const ListItem = ({ server }) => {
  return (
    <div className="list-item" data-testid="list-item">
      <span className="item-text">{server.name}</span>
      <span className="item-text">{`${server.distance} km`}</span>
    </div>
  );
};

export default ListItem;

ListItem.propTypes = {
  server: PropTypes.object
}
