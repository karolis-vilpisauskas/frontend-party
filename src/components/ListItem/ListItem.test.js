import React from "react";
import ReactDOM from "react-dom";
import { cleanup, render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import ListItem from "./ListItem";

afterEach(cleanup);

const createTestProps = () => ({
  server: {
    name: "Server",
    distance: 546
  }
});

it("renders without crashing", () => {
  const props = createTestProps();
  const div = document.createElement("div");
  ReactDOM.render(<ListItem {...props} />, div);
});

it("renders list item correctly", () => {
  const props = createTestProps();
  const { getByTestId } = render(<ListItem {...props} />);
  const container = getByTestId("list-item");
  const serverName = container.firstChild;
  const serverDistance = container.lastChild;

  expect(container.children.length).toBe(2);
  expect(serverName).toHaveTextContent("Server");
  expect(serverDistance).toHaveTextContent("546 km");
});
