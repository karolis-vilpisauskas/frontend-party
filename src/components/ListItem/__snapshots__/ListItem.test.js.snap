// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem rendering it renders without crashing 1`] = `
<div
  class="list-item"
  data-testid="list-item"
>
  <span
    class="item-text"
  >
    Server
  </span>
  <span
    class="item-text"
  >
    546 km
  </span>
</div>
`;
